vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleManager.java

```diff
@@ -11,14 +16,33 @@
 @Service
 public class ModuleManager implements IModuleManager {
 
-	@Autowired
-	private ModuleRepository moduleRepo;
-	
-	/* (non-Javadoc)
-	 * @see edu.asu.diging.vspace.core.services.impl.IModuleManager#storeModule(edu.asu.diging.vspace.core.model.IModule, java.lang.String)
-	 */
-	@Override
-	public IModule storeModule(IModule module) {
-		return moduleRepo.save((Module)module);
-	}
+    @Autowired
+    private ModuleRepository moduleRepo;
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.diging.vspace.core.services.impl.IModuleManager#storeModule(edu.asu.
+     * diging.vspace.core.model.IModule, java.lang.String)
+     */
+    @Override
+    public IModule storeModule(IModule module) {
+        return moduleRepo.save((Module) module);
+    }
+
+    @Override
+    public IModule getModule(String id) {
+        Optional<Module> module = moduleRepo.findById(id);
+        if (module.isPresent()) {
+            try {
+                return module.get();
+            } catch(NoSuchElementException e) {
+                logger.error("No Such Element Exception", e);
+            }            
```

Comment:
you already checked that there is an object present, so no need to catch the exception